src/adapter-in/api/AIApi.ts

```diff
@@ -9,7 +9,7 @@ export interface ListAIsResponse {
 export interface ListAIsRequestParams {
   scope?: ListAIsRequestScope | null;
   groupId?: string | null;
-  categoryId?: string | null;
+  categoryId?: CategoryType |string | null;
```

Comment:
We can change to  
```suggestion
  categoryId?: CategoryType | null;
```


In the API route we can follow the same pattern as with the [scope param](https://github.com/nicolasdesmarais/ai-companion/blob/main/app/api/v1/me/ai/route.ts#L22-L34), which should look like:

```typescript
  const categoryParam = searchParams.get("categoryId");
  let categoryParam: CategoryType | undefined;

  if (
    !categoryParam ||
    !Object.values(CategoryType).includes(
      categoryParam as CategoryType
    )
  ) {
    categoryParam = undefined;
  } else {
    categoryParam = CategoryType[categoryParam as keyof typeof CategoryType];
  }
```

Reply:
Not required now, keeping it as string | null